src/RobotsHeaders.php

```diff
@@ -41,12 +41,50 @@ public function mayFollow(string $userAgent = '*'): bool
 
     public function noindex(string $userAgent = '*'): bool
     {
-        return $this->robotHeadersProperties[$userAgent]['noindex'] ?? false;
+        $userAgent = strtolower($userAgent);
+
+        if (isset($this->robotHeadersProperties[$userAgent])
+            && isset($this->robotHeadersProperties[$userAgent]['noindex'])) {
+            return $this->robotHeadersProperties[$userAgent]['noindex'];
+        }
+
+        if ($wildCard = $this->checkForWildcard($userAgent, 'noindex') !== null) {
+            return $wildCard;
+        }
+
+        return $this->robotHeadersProperties['*']['noindex'] ?? false;
     }
 
     public function nofollow(string $userAgent = '*'): bool
     {
-        return $this->robotHeadersProperties[$userAgent]['nofollow'] ?? false;
+        $userAgent = strtolower($userAgent);
+
+        // check for exact
+        if (isset($this->robotHeadersProperties[$userAgent])
+            && isset($this->robotHeadersProperties[$userAgent]['nofollow'])) {
+            return $this->robotHeadersProperties[$userAgent]['nofollow'];
+        }
+
+        // check for wildcrad user agent
+        if ($wildCard = $this->checkForWildcard($userAgent, 'nofollow') !== null) {
+            return $wildCard;
+        }
+
+        // check for wildcrad
+        return $this->robotHeadersProperties['*']['nofollow'] ?? false;
+    }
+
+    protected function checkForWildcard(string $userAgent, string $property)
+    {
+        if ($userAgent !== '*') {
+            for ($i = 1; $i <= strlen($userAgent); $i++) {
```

Comment:
I don't like loops with temporary variables like this one. The intent of this code isn't clear. Are you check if the string contains a certain string? Create a function with that name and use that.

Reply:
I change the code by creating a new function wich make the intent clearer (i hope), but keep the loop on the temporary variable.

Reply:
The reason for the loop with temporary variables is still unclear to me.

It seems you're looking for the part before `*`. Maybe exploding on `*` could offer a way to avoid the loop?

Reply:
What the code do.
If there are rules for googlebot* or google* and I am searching for googlebot-news.
I check : 
```
g*
go*
goo*
goog*
googl*
google* (may stop if there is google* rules)
googleb*
googlebo*
googlebot* (stop there)
```

Could be change by : 
```

        $wildcardUserAgents = array_filter(array_keys($this->disallowsPerUserAgent), function($userAgent) {
            return strpos($userAgent, '*') !== false;
        });
        foreach ($wildcardUserAgents as $wildcardUserAgent) {
            if (strpos($userAgent, substr($wildcardUserAgent, 0, -1)) === 0)  {
                return $wildcardUserAgent;
            }
        }

```